storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -1647,18 +1660,20 @@ fseg_find_last_used_frag_page_slot(
 /** Calculate reserved fragment page slots.
 @param inode  file segment index
 @return number of fragment pages */
-static ulint fseg_get_n_frag_pages(const fseg_inode_t *inode)
+static ulint fseg_get_n_frag_pages(
+  const fseg_inode_t *inode, range_set *page_ranges= nullptr)
```

Comment:
Instead of adding a parameter to an existing rather commonly used function, I think that it would be better to add a separate function for the rare case that `page_ranges` is specified. That parameter also needs to be documented in a function comment.